aclk/aclk_lws_wss_client.c

```diff
@@ -0,0 +1,315 @@
+#include "aclk_lws_wss_client.h"
+
+#include "libnetdata/libnetdata.h"
+
+static int aclk_lws_wss_callback(struct lws *wsi, enum lws_callback_reasons reason, void *user, void *in, size_t len);
+
+struct aclk_lws_wss_perconnect_data {
+    int todo;
+};
+
+//TODO config??
+int aclk_lws_wss_use_ssl = 1;
+
+struct lws_wss_packet_buffer {
+	char* data;
+	size_t data_size;
+	struct lws_wss_packet_buffer *next;
+};
+
+static inline struct lws_wss_packet_buffer *lws_wss_packet_buffer_new(void* data, size_t size)
+{
+	struct lws_wss_packet_buffer *new = callocz(1, sizeof(struct lws_wss_packet_buffer));
+	if(data) {
+		new->data = malloc(LWS_PRE+size);
+    	memcpy(new->data+LWS_PRE, data, size);
+		new->data_size = size;
+	}
+	return new;
+}
+
+static inline void lws_wss_packet_buffer_push(struct lws_wss_packet_buffer **list, struct lws_wss_packet_buffer *item)
+{
+	struct lws_wss_packet_buffer *tail = *list;
+	if(!*list) {
+		*list = item;
+		return;
+	}
+	while(tail->next) {
+		tail = tail->next;
+	}
+	tail->next = item;
+}
+
+static inline struct lws_wss_packet_buffer *lws_wss_packet_buffer_pop(struct lws_wss_packet_buffer **list)
+{
+	struct lws_wss_packet_buffer *ret = *list;
+	if(ret != NULL)
+		*list = ret->next;
+	//ret->next = NULL;
+	return ret;
+}
+
+static inline void lws_wss_packet_buffer_free(struct lws_wss_packet_buffer *item)
+{
+	free(item->data);
```

Comment:
Considering that you are using `mallocz` to allocate the structure, I think you should use `freez` here instead `free`. At the end they are the same, but it is only to keep a pattern.
No less important, case you keep `malloc` above, please, add an `if (item->data)` before release memory here.